src/components/comments.jsx

```diff
@@ -1,91 +1,92 @@
 import React, { useState, useEffect } from "react";
-import { postComment, getCommentsByArticleID } from "../../utils/api";
+import { postComment, getCommentsByArticleID } from "../utils/api";
 import { useParams } from "react-router-dom";
 import CommentList from "./commentList";
+import { useLogin } from "../contexts/login";
 
 const Comments = ({
   article,
   setCommentCount,
   isCommentListError,
   isLoading,
   comments,
-  setComments
+  setComments,
 }) => {
   const { article_id } = useParams();
   const [commentBody, setCommentBody] = useState("");
-  const [commenter, setCommenter] = useState("");
   const [isPostCommentShown, setIsPostCommentShown] = useState(false);
   const [isEmpty, setIsEmpty] = useState(true);
+  const [hasCommented, setHasCommented] = useState(false)
+
+  const { user } = useLogin();
 
   function handleClickPostComment() {
     setIsPostCommentShown(!isPostCommentShown);
   }
 
   useEffect(() => {
-    if (!commenter || !commentBody) {
+    if (!commentBody) {
       setIsEmpty(true);
     } else setIsEmpty(false);
-  }, [commenter, commentBody]);
+  }, [commentBody]);
 
-  function handleSubmitComment(commenter, commentBody, e) {
+  function handleSubmitComment(commentBody, e) {
     e.preventDefault();
-    return postComment(commenter, commentBody, article_id).then(() => {
-      setCommentCount(article.comment_count)
-      getCommentsByArticleID(article_id).then(({data}) => {
-        setComments(data.comments)
-      })
-    })
+    return postComment(user, commentBody, article_id).then(() => {
+      setCommentCount(article.comment_count);
+      getCommentsByArticleID(article_id).then(({ data }) => {
```

Comment:
No nested .thens please!

Reply:
If you made use of optimistic rendering, then there would also be no need to re-fetch the comments by article id. Re-setting the list of comments would cause a re-render of the initial comment fetch